cram_pr2/cram_pr2_environment_manipulation/src/plans.lisp

```diff
@@ -0,0 +1,166 @@
+  (exe:perform
+   (desig:an action
+             (type setting-gripper)
+             (gripper ?arm)
+             (position 0)))
+  (roslisp:ros-info (environment-manipulation open-container) "Opening")
+  (when (and joint-name environment)
+    (cram-occasions-events:on-event
+     (make-instance 'cpoe:container-handle-grasping-event
+                    :joint-name joint-name
+                    :side ?arm
+                    :environment environment)))
+  (let ((?left-lift-pose (list (first ?left-lift-poses)))
+        (?left-2nd-lift-pose (list (second ?left-lift-poses)))
+        (?right-lift-pose (list (first ?right-lift-poses)))
+        (?right-2nd-lift-pose (list (second ?right-lift-poses))))
+    (exe:perform
+     (desig:an action
+               (type lifting)
+               (left-poses ?left-lift-pose)
+               (right-poses ?right-lift-pose)))
+    (when (and joint-name environment)
+      (cram-occasions-events:on-event
+       (make-instance 'cpoe:container-opening-event
+                      :joint-name joint-name
+                      :side ?arm
+                      :environment environment)))
+    (exe:perform
+     (desig:an action
+               (type setting-gripper)
+               (gripper ?arm)
+               (position 0.1)))
+    (exe:perform
+     (desig:an action
+               (type lifting)
+               (left-poses ?left-2nd-lift-pose)
+               (right-poses ?right-2nd-lift-pose))))
+  (when (btr:robot-colliding-objects-without-attached)
+    (error 'common-fail:manipulation-pose-in-collision)))
+
+(defun close-container (?arm ?gripper-opening
+                        ?left-reach-poses ?right-reach-poses
+                        ?left-lift-poses ?right-lift-poses
+                        &optional
+                          (joint-name nil)
+                          (environment nil))
+  (cpl:par
+    (roslisp:ros-info (environment-manipulation close-container) "Opening gripper")
+    (exe:perform
+     (desig:an action
+               (type setting-gripper)
+               (gripper ?arm)
+               (position ?gripper-opening)))
+    (roslisp:ros-info (environment-manipulation close-container) "Reaching")
+    (exe:perform
+     (desig:an action
+               (type reaching)
+               (left-poses ?left-reach-poses)
+               (right-poses ?right-reach-poses))))
+  (roslisp:ros-info (environment-manipulation close-container) "Gripping")
+  (exe:perform
+   (desig:an action
+             (type setting-gripper)
+             (gripper ?arm)
+             (position 0)))
+  (roslisp:ros-info (environment-manipulation close-container) "Closing")
+  (when (and joint-name environment)
+    (cram-occasions-events:on-event
+     (make-instance 'cpoe:container-handle-grasping-event
+                    :joint-name joint-name
+                    :side ?arm
+                    :environment environment)))
+  (let ((?left-lift-pose (list (first ?left-lift-poses)))
```

Comment:
Please separate these in -designators.lisp file, data management should not happen in plans. Only control flow.